pkg/combustion/elemental.go

```diff
@@ -0,0 +1,88 @@
+package combustion
+
+import (
+	_ "embed"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/suse-edge/edge-image-builder/pkg/fileio"
+	"github.com/suse-edge/edge-image-builder/pkg/image"
+	"github.com/suse-edge/edge-image-builder/pkg/log"
+	"github.com/suse-edge/edge-image-builder/pkg/template"
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	elementalComponentName = "elemental"
+	elementalScriptName    = "11-elemental.sh"
+	elementalConfigName    = "elemental_config.yaml"
+)
+
+//go:embed templates/11-elemental-register.sh.tpl
+var elementalScript string
+
+func configureElemental(ctx *image.Context) ([]string, error) {
+	// Even if the "elemental" section is left out of the definition, the full structure will
```

Comment:
What do we mean here? I get that the criteria for the configuration is a non-empty registration URL (until a validation step is implemented), however we don't (and also shouldn't) modify any of the section values.

Reply:
Ok, maybe I need to reword this. It was an explanation for developers potentially not familiar with how the yaml parsing works.

My gut reaction was to check if `Registration` or `Elemental` were nil. But the way the parser works, it will create them with the default values. So it was meant to explain the rationale for checking a specific value instead of just the definition structure itself. Perhaps it's overkill though; maybe it's obvious to new devs that checking for the URL makes sense as the logic.